### ISSUE TITLE
Determine formats for fundamental lesson comprehension questions

### ISSUE THREAD

**Issue by elPiablo** [Developer-DAO/academy#147]
We need:
A set of questions will let the learner know if they have actually understood what a fundamental has offered to teach.

**Creating a set of questions will also give a content writer a clearer picture of what they want a learner to actually learn, and therefore become an autonomous learner, as opposed to just replicate what e.g. code in a lesson does and become.......a copy cat, having not actually learned anything.**

Not all lessons follow the same format. The question is, should all sets of questions?

How would we do this with regards to short and long term memory retention of the content we are teaching?
We can include questions periodically throughout a lesson, especially if it is quite a long one. And/or we can include a set of questions at the end.

It is important for motivation purposes that a learner can answer a good amount of questions. I propose grouping questions. The following two ways are a couple of examples, and there are of course more possible groupings.

a) sets of general questions on easy, medium, difficult to answer [dev: in an array]
b) questions grouped into topics e.g. programming language, developer tooling, protocol information, expected outputs [dev: in an array]

Content submission:
As many as possible questions and answers provided by the content provider could solve the problem and also ensure to Academy platform that the answers to the questions are indeed correct. And cut out unnecessary reviewing The more questions/answers we have, the less chance that a user might publish them to others *to potentially cheat*. We can also explicitly ask that they don't, at least not in bulk!!

Development:
A Javascript nested loop could be the function we need to solve this. We put the questions and answers in arrays - could we automate this process in some way for the content writer to do? Using some kind of form that will automatically script this process? @7i7o ?

Would we use a simple form with multiple choice in the lesson body? Or is there a tool we can use to get a better standard of Q and A? Would this go into a side drawer? I think a certain score e.g. 80%, should be a compulsory measure for the learner to be able to advance. And can this be linked to a user's wallet to sign with their result?

Would the same experience be able to be replicated on a desktop and a mobile in the same way?

**Comment by elPiablo** [comment]
Not sure if there is a need for this at the moment.  

We are already incorporating tests into lesson bodies (https://github.com/Developer-DAO/academy/issues/146). But the future may be different, depending on content created in the DAO, or externally. e.g. @jello-git3 is creating a fundamentals set of lessons on crytography. 

Considerations on the implementation of an LMS (learning management system) will also factor in making a decision/choice on this isssue. 

**In summary**:  I think we should keep this Issue open for the moment.

**Comment by elPiablo** [comment]
It took a while, but we can close this issue as it has been resolved.
Kudos to everyone for their work on this.
Testing users on the knowledge that they acquire is a really important feature of the Academy, and anyone's learning journey.
Big shout out to Academy team past and present for all your thoughts on this.
Bigger shout out to Mveve, @7i7o and @georgemac510 for for getting our first batch of tests out the door, therefore laying down these 'good education practices' to the wider web3 educators community.
Shout from the rooftops to @AlanRacciatti for coding an app for non-technical folks can add their own user knowledge tests/quizzes easily to their education content!!
You all made this process fun and enriching.
